packages/polar/src/internal/core/config/config-validation.ts

```diff
@@ -0,0 +1,111 @@
+import * as z from 'zod';
+
+import { parseZodError } from "../../util/zod-errors";
+//import { validateAccount } from "../../../lib/account";
+import type { NetworkConfig } from "../../../types";
+import CfgErrors from "./config-errors";
+import { PolarError } from '../errors';
+import { ERRORS } from '../errors-list';
+
+const AccountType = z.object({
+  name: z.string(),
+  address: z.string(),
+  mnenomic: z.string()
+});
+
+const HttpHeaders = z.record(z.string());
+
+const HttpNetworkType = z.object({
+  accounts: z.array(AccountType).optional(),
+  endpoint: HttpHeaders.optional(),
+  nodeId: z.string().optional(),
+  chainId: z.string().optional(),
+  keyringBackend: z.string().optional()
+}).nonstrict();
+
+const NetworksType = z.record(HttpNetworkType);
+
+const ProjectPaths = z.object({
+  root: z.string().optional(),
+  cache: z.string().optional(),
+  artifacts: z.string().optional(),
+  sources: z.string().optional(),
+  tests: z.string().optional()
+}).nonstrict(); ;
+
+const Config = z.object({
+  networks: NetworksType.optional(),
+  paths: ProjectPaths.optional()
+}
+).nonstrict(); ;
+
+/**
+ * Validates the config, throwing a BuilderError if invalid.
+ * @param config
+ */
+export function validateConfig(config: any) { // eslint-disable-line
+  const errors = getValidationErrors(config);
+
+  if (errors.isEmpty()) {
+    return;
+  }
+
+  const errorList = `  * ${errors.toString()}`;
+  throw new PolarError(ERRORS.GENERAL.INVALID_CONFIG, { errors: errorList });
+}
+
+export function getValidationErrors(config: any): CfgErrors {  // eslint-disable-line
```

Comment:
Replace type here from `any`.

Reply:
this should be any because we are validating types in this function, so config passed could be any but it will be checked for PolarConfig

Reply:
okay